public/app/features/dashboard/components/SaveDashboard/SaveDashboardDrawer.tsx

```diff
@@ -0,0 +1,175 @@
+  const hasVariableChanged = useMemo(() => dashboard.hasVariableValuesChanged(), [dashboard]);
+
+  const [saveTimerange, setSaveTimerange] = useState(false);
+  const [saveVariables, setSaveVariables] = useState(false);
+
+  const status = useMemo(() => {
+    const isProvisioned = dashboard.meta.provisioned;
+    const isNew = dashboard.version === 0;
+    const isChanged = dashboard.version > 0;
+
+    return {
+      isProvisioned,
+      isNew,
+      isChanged,
+    };
+  }, [dashboard]);
+
+  const previous = useAsync(async () => {
+    if (status.isNew) {
+      return undefined;
+    }
+
+    const result = await backendSrv.getDashboardByUid(dashboard.uid);
+    return result.dashboard;
+  }, [dashboard, status]);
+
+  const data = useMemo<SaveDashboardData>(() => {
+    const clone = dashboard.getSaveModelClone({
+      saveTimerange: Boolean(saveTimerange),
+      saveVariables: Boolean(saveVariables),
+    });
+
+    if (!previous.value) {
+      return { clone, diff: {}, diffCount: 0, hasChanges: false };
+    }
+
+    const cloneJSON = JSON.stringify(clone, null, 2);
+    const cloneSafe = JSON.parse(cloneJSON); // avoids undefined issues
+
+    const diff = jsonDiff(previous.value, cloneSafe);
+    let diffCount = 0;
+    for (const d of Object.values(diff)) {
+      diffCount += d.length;
+    }
+
+    return {
+      clone,
+      diff,
+      diffCount,
+      hasChanges: diffCount > 0 && !status.isNew,
+    };
+  }, [dashboard, previous.value, saveTimerange, saveVariables, status.isNew]);
+
+  const [showDiff, setShowDiff] = useState(false);
+  const { state, onDashboardSave } = useDashboardSave(dashboard);
+  const onSuccess = onSaveSuccess
+    ? () => {
+        onDismiss();
+        onSaveSuccess();
+      }
+    : onDismiss;
+
+  const renderBody = () => {
+    if (showDiff) {
+      return <SaveDashboardDiff diff={data.diff} oldValue={previous.value} newValue={data.clone} />;
+    }
+
+    if (status.isNew || isCopy) {
+      return (
+        <SaveDashboardAsForm
+          dashboard={dashboard}
+          onCancel={onDismiss}
+          onSuccess={onSuccess}
+          onSubmit={onDashboardSave}
+          isNew={status.isNew}
+        />
+      );
+    }
+
+    if (status.isProvisioned) {
+      return <SaveProvisionedDashboardForm dashboard={dashboard} onCancel={onDismiss} onSuccess={onSuccess} />;
+    }
+
+    return (
+      <SaveDashboardForm2
+        dashboard={dashboard}
+        saveModel={data}
+        onCancel={onDismiss}
+        onSuccess={onSuccess}
+        onSubmit={onDashboardSave}
+        options={{
+          saveTimerange,
+          saveVariables,
+        }}
+      />
+    );
+  };
+
+  if (state.error) {
+    return (
+      <SaveDashboardErrorProxy
+        error={state.error}
+        dashboard={dashboard}
+        dashboardSaveModel={data.clone}
+        onDismiss={onDismiss}
+      />
+    );
+  }
+
+  return (
+    <Drawer
+      title={dashboard.title}
```

Comment:
Having just the dashboard name as title feels unintuitive here, should be the action that is the title (Ie "Save dashboard" , or "Save as copy" etc.). I tested to add dashboard title as subtitle but not sure we need it. Maybe we can add Name as an option in the Options tab (like it is under Save as form).